admin-ui/src/component/my-units/MyUnitRecurringReservation/MyUnitRecurringReservation.tsx

```diff
@@ -0,0 +1,104 @@
+import { useQuery } from "@apollo/client";
+import { H1 } from "common/src/common/typography";
+import {
+  Query,
+  QueryUnitsArgs,
+  ReservationUnitType,
+} from "common/types/gql-types";
+import { Button, IconAngleLeft } from "hds-react";
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useParams } from "react-router-dom";
+import styled from "styled-components";
+import { myUnitUrl } from "../../../common/urls";
+import { useNotification } from "../../../context/NotificationContext";
+import { Container } from "../../../styles/layout";
+import { BasicLink } from "../../../styles/util";
+import Loader from "../../Loader";
+import withMainMenu from "../../withMainMenu";
+import { RECURRING_RESERVATION_UNIT_QUERY } from "../queries";
+import { MyUnitRecurringReservationForm } from "./MyUnitRecurringReservationForm";
+import type { ReservationsMade } from "./RecurringReservationDone";
+import RecurringSuccess from "./RecurringReservationDone";
+
+const PreviousLinkWrapper = styled.div`
+  padding: var(--spacing-xs);
+`;
+
+type Params = {
+  unitId: string;
+  reservationUnitId: string;
+};
+
+const MyUnitRecurringReservation = () => {
+  const [sent, setSent] = useState<ReservationsMade | null>(null);
+  const { notifyError } = useNotification();
+  const { t } = useTranslation();
+  // FIXME maybe params need better handling
+  const { unitId } = useParams<Params>();
+
+  const previousUrl = myUnitUrl(parseInt(unitId ?? "0", 10));
+  const { loading, data: unitData } = useQuery<Query, QueryUnitsArgs>(
+    RECURRING_RESERVATION_UNIT_QUERY,
+    {
+      variables: {
+        pk: [unitId ?? "0"],
+        offset: 0,
+      },
+      onError: (err) => {
+        notifyError(err.message);
+      },
+    }
+  );
+  const unit = unitData?.units?.edges[0];
+  const reservationUnits = unit?.node?.reservationUnits?.filter(
+    (item): item is ReservationUnitType => !!item
+  );
+
+  const handleReservation = (res: ReservationsMade) => {
+    setSent(res);
+  };
+
+  if (loading) return <Loader />;
+
+  return (
+    <>
+      <PreviousLinkWrapper>
+        <BasicLink to={previousUrl}>
+          <Button
+            aria-label={t("common.prev")}
+            size="small"
+            variant="supplementary"
+            iconLeft={<IconAngleLeft />}
+          >
+            {t("common.prev")}
+          </Button>
+        </BasicLink>
+      </PreviousLinkWrapper>
+
+      <Container>
+        {sent != null ? (
```

Comment:
Not for null checks since
``` 
(undefined === null) === false // but 
(undefined == null) === true
```

Reply:
The problem with loose equality operators is that it's pretty close to a typo, if using a strict operator would be viable. In cases like these, where you check against general falsiness, I'd equate to booleans (`!sent`).

Reply:
Don't like !sent because.
```
 !"" === true 
!0 === true
```

Linter can't catch these, sometimes "" is valid where null isn't, that's the exact case where you'd do something like 
```
const [var, setVar] = useState<string | null>(null)
```

Linter should be used to catch typos i.e. we should use one of these in eslint
eqeqeq: ["error", "smart"]
eqeqeq: ["error", "always", { null: "never" }]